handshake/connection_parameters_manager.go

```diff
@@ -61,14 +62,19 @@ var (
 )
 
 // NewConnectionParamatersManager creates a new connection parameters manager
-func NewConnectionParamatersManager(pers protocol.Perspective, v protocol.VersionNumber) ConnectionParametersManager {
+func NewConnectionParamatersManager(pers protocol.Perspective, v protocol.VersionNumber, maxIncomingDynamicStreamsPerConnection uint32) ConnectionParametersManager {
+	if maxIncomingDynamicStreamsPerConnection == 0 {
```

Comment:
I'd prefer to move this logic to the populate config functions (similar to your other PR), so that the config has valid values.